src/Systems.jl

```diff
@@ -70,22 +70,57 @@ end
 
 
 """
-    SpinSystem(crystal::Crystal, ints::Vector{<:Interaction}, latsize, S=1)
+    propagate_site_info(cryst::Crystal, sites_info::Vector{SiteInfo})
+
+Given an incomplete list of site information, propagates spin magnitudes
+and symmetry-transformed g-tensors to all symmetry-equivalent sites. Throws
+an error if two symmetry-equivalent sites are provided in `sites_info`.
+"""
+function propagate_site_info(crystal::Crystal, sites_info::Vector{SiteInfo})
+    # All sites not explicitly provided are by default S = 1, g = 1
+    all_sites_info = [SiteInfo(i, 1, 1) for i in 1:nbasis(crystal)]
+
+    specified_atoms = Int[]
+    for siteinfo in sites_info
+        @unpack site, S, g = siteinfo
+        (sym_bs, sym_gs) = all_symmetry_related_couplings(crystal, Bond(site, site, [0,0,0]), g)
+        for (sym_b, sym_g) in zip(sym_bs, sym_gs)
```

Comment:
Maybe here we need to be especially careful about the `g` vs `g'` convention?

Reply:
Yeah, this is a place to be careful. The convention I'm using everywhere is

`m_i = g s_i`

where `g` is applied to the left. Under this convention, I believe this is the correct way to transform since exchange matrices also are applied on the left.

Reply:
Nice